### PR TITLE
Drop Kanban development process, rework common labels

### PR DESCRIPTION
**Pull request by ChristophWurst**
The Kanban dev process was used many years ago but never in Nextcloud.
Therefore the process description is not just outdated but misleading.

We still use the labels, albeit their description was mildly outdated.
They were restructured and grouped for better comprehensibility.

Ref https://github.com/nextcloud/groupware/issues/9

Bildschirmfoto von 2021-12-28 16-10-58

### CODE REVIEW

File: developer_manual/getting_started/codingguidelines.rst [hunk 1]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+
+* Tags showing the type of issue or PR
+
+  * ``bug`` - this issue is a bug
+  * ``enhancement`` - this issue is a feature request/idea for improvement of Nextcloud
+  * ``technical debt`` - this issue or PR is about `technical debt <https://en.wikipedia.org/wiki/Technical_debt>`_
+
+* Tags that classify an issue or PR
+
+  * ``high``, ``medium`` ``low`` signify how important the bug is.
```
Comment (splitt3r) [inline developer_manual/getting_started/codingguidelines.rst:44]:
```suggestion
  * ``high``, ``medium``, ``low`` signify how important the bug is.
```

File: developer_manual/getting_started/codingguidelines.rst [hunk 2]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
-* #high #medium #low signify how important the bug is.
 * Tags showing the state of the issue or PR, numbered 0-4:
 
-  * 0\. Needs triage - ready to start development on this
-  * 1\. to develop - ready to start development on this
-  * 2\. developing - development in progress
-  * 3\. to review - ready for review
-  * 4\. to release - reviewed PR that awaits unfreeze of a branch to get merged
-* Feature tags: #feature: something. These tags indicate the features across apps and components which are impacted by the issue or which the PR is related to
+  * ``0. to triage`` - ready to start development on this
```
Comment (juliushaertl) [inline developer_manual/getting_started/codingguidelines.rst:29]:
```suggestion
  * ``0. to triage`` - issue or feature request needs to get triaged and approved for development
```

Open to other suggestions on the description but should be different from "1. to develop"

File: developer_manual/getting_started/codingguidelines.rst [hunk 3]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+  * ``1. to develop`` - ready to start development on this
+  * ``2. developing`` - development in progress
+  * ``3. to review`` - ready for review
+  * ``4. to release`` - reviewed PR that awaits unfreeze of a branch to get merged
```
Comment (juliushaertl) [inline developer_manual/getting_started/codingguidelines.rst:33]:
```suggestion
  * ``4. to release`` - reviewed PR that awaits unfreeze of a branch to get merged or has pending CI jobs
```

File: developer_manual/getting_started/codingguidelines.rst [hunk 4]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+
+* Tags that classify an issue or PR
+
+  * ``high``, ``medium`` ``low`` signify how important the bug is.
+  * ``regression`` - something that worked in a previous release but is now not working as expected or missing.
+  * ``feature: *``, e.g. ``feature: dav``. These tags group tickets of specific feature or subsystems.
+  * ``design`` - this needs help from the design team or is a design-related issue/pull request
+  * ``good first issue`` - these are issues which are relatively easy to solve and ideal for people who want to learn how to code in Nextcloud
+
+* ``backport-request`` - the pull requests also needs to be applied to older Nextcloud versions. This tag is typically assigned by automation.
```
Comment (splitt3r) [inline developer_manual/getting_started/codingguidelines.rst:50]: Is the automation / bot documented somewhere? Maybe a link or a sentence or two on how it works?

Comment (ChristophWurst) [reply developer_manual/getting_started/codingguidelines.rst:50]: As so many things: no, it's not documented at all, so I raised https://github.com/nextcloud/documentation/issues/7881

File: developer_manual/getting_started/codingguidelines.rst [hunk 5]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+
+  * ``high``, ``medium`` and ``low`` signify how important the bug is.
+  * ``regression`` - something that worked in a previous release but is now not working as expected or missing.
+  * ``feature: *``, e.g. ``feature: dav``. These tags group tickets of specific feature or subsystems.
+  * ``design`` - this needs help from the design team or is a design-related issue/pull request
+  * ``good first issue`` - these are issues which are relatively easy to solve and ideal for people who want to learn how to code in Nextcloud
+
+* ``backport-request`` - the pull requests also needs to be applied to older Nextcloud versions. This tag is typically assigned by automation.
 
 If you want a label not in the list above, please first discuss on the mailing list.
```
Comment (splitt3r) [inline developer_manual/getting_started/codingguidelines.rst:52]: Does not make sense anymore (https://github.com/nextcloud/documentation/pull/7879)?

File: developer_manual/getting_started/codingguidelines.rst [hunk 6]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+
+* Tags showing the type of issue or PR
+
+  * ``bug`` - this issue is a bug
+  * ``enhancement`` - this issue is a feature request/idea for improvement of Nextcloud
+  * ``technical debt`` - this issue or PR is about `technical debt <https://en.wikipedia.org/wiki/Technical_debt>`_
+
+* Tags that classify an issue or PR
+
+  * ``high``, ``medium`` and ``low`` signify how important the bug is.
```
Comment (splitt3r) [inline developer_manual/getting_started/codingguidelines.rst:44]:
```suggestion
  * ``high``, ``medium`` and ``low`` - signify how important the bug is
```
To unify the list?

File: developer_manual/getting_started/codingguidelines.rst [hunk 7]
```diff
@@ -24,22 +24,30 @@ We assign labels to issues and pull requests to make it easy to find them and to
...
+  * ``regression`` - something that worked in a previous release but is now not working as expected or missing.
+  * ``feature: *``, e.g. ``feature: dav``. These tags group tickets of specific feature or subsystems.
```
Comment (splitt3r) [inline developer_manual/getting_started/codingguidelines.rst:46]:
```suggestion
  * ``feature: *``, e.g. ``feature: dav`` - these tags group tickets of specific feature or subsystems
```
Maybe?

**Review by splitt3r — APPROVED** [review]

**Pull request merged**